service/tier2.go

```diff
@@ -499,20 +492,18 @@ excludable:
 
 	var streamErr error
 	if canSkipBlockSource(executionPlan.ExistingExecOuts, executionPlan.RequiredModules, request.BlockType) {
-		maxDistributorLength := int(stopBlock - requestDetails.ResolvedStartBlockNum)
-		clocksDistributor := make(map[uint64]*pbsubstreams.Clock)
-		for _, execOutput := range executionPlan.ExistingExecOuts {
-			execOutput.ExtractClocks(clocksDistributor)
-			if len(clocksDistributor) >= maxDistributorLength {
-				break
-			}
-		}
 
-		sortedClocksDistributor := sortClocksDistributor(clocksDistributor)
 		ctx, span := reqctx.WithSpan(ctx, "substreams/tier2/pipeline/mapper_stream")
-		for _, clock := range sortedClocksDistributor {
-			if clock.Number < startBlock || clock.Number >= stopBlock {
-				panic("reading from mapper, block was out of range") // we don't want to have this case undetected
+
+		distributor := execout.NewClockDistributor(executionPlan.ExistingExecOuts, startBlock, stopBlock)
```

Comment:
Same here about Go iterators ..., would normally allow for `for (x, y) in range distributor { ... }`